src/main/java/org/jboss/aesh/readline/actions/Complete.java

```diff
@@ -49,11 +50,12 @@ public void apply(InputProcessor inputProcessor) {
             }
             else {
                 inputProcessor.getCompleter().setAskDisplayCompletion(false);
-                inputProcessor.getBuffer().getUndoManager().clear();
-                inputProcessor.getBuffer().out().print(Config.getLineSeparator());
-                inputProcessor.clearBufferAndDisplayPrompt();
-                inputProcessor.getBuffer().drawLine();
-                inputProcessor.getBuffer().moveCursor(inputProcessor.getBuffer().getBuffer().getMultiCursor());
+                ConsoleBuffer buffer = inputProcessor.getBuffer();
+                buffer.getUndoManager().clear();
+                buffer.out().print(Config.getLineSeparator());
+                buffer.displayPrompt();
+                buffer.out().print(buffer.getBuffer().getLine());
+                buffer.moveCursor(buffer.getBuffer().getMultiCursor());
```

Comment:
I didn't use `{{inputProcessor.getBuffer().drawLine(); `, as it placed a cursor at the beginning of the line, instead of at the end. In the original code it also seems redundant with previous line `inputProcessor.clearBufferAndDisplayPrompt();`.